egs/wsj/s5/steps/nnet3/tdnn/train_raw.sh

```diff
@@ -0,0 +1,636 @@
+  echo "                                                   # use multiple threads... note, you might have to reduce mem_free,ram_free"
+  echo "                                                   # versus your defaults, because it gets multiplied by the -pe smp argument."
+  echo "  --minibatch-size <minibatch-size|128>            # Size of minibatch to process (note: product with --num-threads"
+  echo "                                                   # should not get too large, e.g. >2k)."
+  echo "  --samples-per-iter <#samples|400000>             # Number of samples of data to process per iteration, per"
+  echo "                                                   # process."
+  echo "  --splice-indexes <string|layer0/-4:-3:-2:-1:0:1:2:3:4> "
+  echo "                                                   # Frame indices used for each splice layer."
+  echo "                                                   # Format : layer<hidden_layer_index>/<frame_indices>....layer<hidden_layer>/<frame_indices> "
+  echo "                                                   # (note: we splice processed, typically 40-dimensional frames"
+  echo "  --lda-dim <dim|''>                               # Dimension to reduce spliced features to with LDA"
+  echo "  --realign-times <list-of-times|\"\">             # A list of space-separated floating point numbers between 0.0 and"
+  echo "                                                   # 1.0 to specify how far through training realignment is to be done"
+  echo "  --align-cmd (utils/run.pl|utils/queue.pl <queue opts>) # passed to align.sh"
+  echo "  --align-use-gpu (yes/no)                         # specify is gpu is to be used for realignment"
+  echo "  --num-jobs-align <#njobs|30>                     # Number of jobs to perform realignment"
+  echo "  --stage <stage|-4>                               # Used to run a partially-completed training process from somewhere in"
+  echo "                                                   # the middle."
+
+
+  exit 1;
+fi
+
+data=$1
+targets_scp=$2
+dir=$3
+
+# Check some files.
+for f in $data/feats.scp $targets_scp; do
+  [ ! -f $f ] && echo "$0: no such file $f" && exit 1;
+done
+
+if $add_final_sigmoid && $include_log_softmax; then
+  echo "add-final-sigmoid and include-log-softmax cannot both be true"
+fi
+
+if $dense_targets; then
+  # For raw nnet, read target dimension from the targets_scp file
+  num_targets=`feat-to-dim scp:$targets_scp - 2>/dev/null` || exit 1
+fi
+
+[ -z "$num_targets" ] && echo "\$num_targets is unset" && exit 1
+[ "$num_targets" -eq "0" ] && echo "\$num_targets is 0" && exit 1
+
+# in this dir we'll have just one job.
+sdata=$data/split$nj
+utils/split_data.sh $data $nj
+
+mkdir -p $dir/log
+echo $nj > $dir/num_jobs
+
+
+# First work out the feature and iVector dimension, needed for tdnn config creation.
+case $feat_type in
+  raw) feat_dim=$(feat-to-dim --print-args=false scp:$data/feats.scp -) || \
```

Comment:
while you are at it you could drop support for lda features from this script to make it cleaner. We are not using this option in nnet3 scripts as we use a preconditioning matrix in the network.